### DISCUSSION TITLE
Is this possible to run in a Jupyter Notebook?

### DISCUSSION THREAD

**Discussion by tcbentley** [mattdodge/yahoofantasy#26] [Q&A]
And if so, are there any changes that need to be made to my files or the way I launch?

When I run the following in a notebook (after successfully logging in via `yahoofantasy login` in terminal):

```
from yahoofantasy import Context
ctx = Context()
leagues = ctx.get_leagues('mlb', 2020)
for league in leagues:
    print(league.name + " -- " + league.league_type)
```

I get the below response:
`ValueError: Client ID, secret, and refresh token are required. Did you run 'yahoofantasy login' already?`

When I run the same code in my terminal after the successful login, I get:
`from: can't read /var/mail/yahoofantasy`

Finally, if I prefix the code in terminal with `python3 `, I get:
`[Errno 2] No such file or directory`

Any ideas on what I'm doing wrong? Thanks for any help!

**Comment by mattdodge** [comment]
This is a great question, something I hadn't considered before. The short answer is I'm not sure, I'm only minorly familiar with jupyter notebooks. I can give you some context as to how it works though which may help.

When you run the `yahoofantasy login` command it actually creates a `.yahoofantasy` file in that same directory. This file contains your Yahoo authentication tokens as well as any other persisted/cached API responses. When you create a `Context` object it attempts to read from this file to get your Yahoo API access credentials. If it can't read the file or there are no credentials there you will see the ValueError that you posted above.

So I think the key here is to just make sure that the file that is created when you run the login command is also available to the jupyter notebook. I'm not exactly sure how file persistence and access works within notebooks like that, but hopefully that points you in the right direction?
